citesphere/src/test/java/edu/asu/diging/citesphere/core/service/impl/CitationManagerTest.java

```diff
@@ -75,7 +83,7 @@
     public void setUp() throws ZoteroHttpStatusException {
         MockitoAnnotations.initMocks(this);
         managerToTest.init();
-        
+        managerToTestSpy  = Mockito.spy(managerToTest);
```

Comment:
what is this for? Why are you not using `managerToTest`?

Reply:
I need to mock a function which is defined in CitationManager, If I use managerToTest then it is not able to mock it. On google search, I found we need to create spy object if we want to call function defined in the same testing class.

Reply:
instead of mocking methods in the same class you're testing, mock it's dependencies and get rid of the spy.